File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefix = []
        count =0
        for num in nums:
            count+=num
            self.prefix.append(count)

    def sumRange(self, left: int, right: int) -> int:
        leftSum = self.prefix[left-1] if left > 0 else 0
        rightSum = self.prefix[right]
        return rightSum - leftSum
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.java
class NumArray {
    private int[] prefix;
    public NumArray(int[] nums) {
        prefix = new int[nums.length];
        int count = 0;
        for(int i=0; i<nums.length; i++){
            count += nums[i];
            prefix[i] = count;
        }
    }
    
    public int sumRange(int left, int right) {
        int leftSum = 0;
        if (left > 0){
            leftSum += prefix[left-1];
        }
        int rightSum = prefix[right];
        return rightSum - leftSum;

    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */